docs/decisions/0055-dotnet-azureopenai-stable-version-strategy.md

```diff
@@ -0,0 +1,224 @@
+- Our package versioning should make it clear which version of OpenAI or Azure.AI.OpenAI packages we depend on
+- Follow the Semantic Kernel versioning strategy in a way that accommodates well with other SDK version strategies.
+
+## Considered Options
+
+1. **Keep As-Is** - Target only preview packages.
+2. **Preview + GA versioning** (Create a new version (GA + pre-release) side by side of the Azure OpenAI and OpenAI Connectors).
+3. **Stop targeting preview packages**, only target GA packages moving forward.
+
+## 1. Keep As-Is - Target only preview packages
+
+This option will keep the current strategy of targeting only preview packages, which will keep the compatibility with the previous versions and new GA targeting versions and pipelines for our customers. This option has the least impact on our users and our pipeline strategy.
+
+Today all customers that are already using Azure OpenAI Connector have their pipelines configured to use the preview packages.
+
+```mermaid
+%%{init: { 'logLevel': 'debug', 'theme': 'base', 'gitGraph': {'showBranches': true, 'showCommitLabel':true,'mainBranchName': 'SemanticKernel'}} }%%
+      gitGraph TB:
+        checkout SemanticKernel
+        commit id:"SK 1.21"
+        branch OpenAI
+        commit id:"OAI 2.0-beta.12"
+        branch AzureOpenAI
+        commit id:"AOAI 2.0-beta.6"
+        checkout SemanticKernel
+        merge OpenAI id:"SK OAI 1.22"
+        merge AzureOpenAI id:"SK AOAI 1.22"
+        checkout OpenAI
+        commit id:"OAI 2.0 GA"
+        checkout AzureOpenAI
+        merge OpenAI id:"AOAI 2.0 GA"
+        checkout SemanticKernel
+        commit id:"Skipped GA's"
+        checkout OpenAI
+        commit id:"OAI 2.1-beta.1"
+        checkout AzureOpenAI
+        commit id:"AOAI 2.1-beta.1"
+        checkout SemanticKernel
+        merge OpenAI id:"SK OAI 1.23"
+        merge AzureOpenAI id:"SK AOAI 1.23"
+```
+
+Pros:
+
+- No changes in strategy. (Least impact on customers)
+- Keep the compatibility with the previous versions and new GA targeting versions and pipelines.
+- Compatible with our previous strategy of targeting preview packages.
+- Azure and OpenAI SDKs will always be in sync with new GA versions, allowing us to keep the targeting preview with the latest GA patches.
+
+Cons:
+
+- There won't be a SK connector version that targets a stable GA package for OpenAI or AzureOpenAI.
+- New customers that understand and target GA only available features and also have a strict requirement for dependent packages to be also GA will not be able to use the SK connector. (We don't have an estimate but this could be very small compared to the number of customers that are already OK on using the preview Azure SDK OpenAI SDK available for the past 18 months)
+- Potential unexpected breaking changes introduced by OpenAI and Azure.AI.OpenAI beta versions that eventually we might be passing on due to their dependency.
+
+## 2. Preview + GA versioning
```

Comment:
Would a "GA" package for `Microsoft.SemanticKernel.Agents.OpenAI` even be published?  It would be unusable for "GA" as it is entirely based on what is currently a "preview" feature?

Reply:
This is a preview feature for `Azure OpenAI SDK` that didn't make to GA package, but for OpenAI GA it was made available as `Experimental` feature.

https://github.com/openai/openai-dotnet/blob/c49dd7065215bc0d094c7f79ccd634a38f0d7b66/src/Custom/Assistants/AssistantClient.cs#L14C1-L14C28